### PR TITLE
implemented test and Changed assert to TORCH_CHECK #88808

### LINKED ISSUE #88808: `median` INTERNAL ASSERT FAILED with `dims < 25` on CUDA
### 🐛 Describe the bug

`median` INTERNAL ASSERT FAILED with `dims < 25` on CUDA. By contrast, it works well on cpu

```py
import torch
import numpy as np

input_data = torch.Tensor(np.random.uniform(size=(2, 3))).cuda()
output = torch.tensor([[[[[[[[[[[[[[[[[[[[[[[[2.3200, 4.1619, 4.2059],
                              [3.3720, 2.7839, 3.6389]]]]]]]]]]]]]]]]]]]]]]]],device='cuda:0')
input = torch.tensor([[[[[[[[[[[[[[[[[[[[[[[[[   float('nan'),    float('nan'),    float('nan')],
                               [   float('nan'),    float('nan'), 0.6918]]]]]]]]]]]]]]]]]]]]]]]]],device='cuda:0')
output = torch.mean(torch.abs((output - input)), dim=(- 1), keepdim=True)
output = torch.mean(torch.abs((output - input_data)), dim=(- 1), keepdim=True)
output = torch.mean(torch.abs((output - torch.min(output))), dim=(- 1), keepdim=True)
output = torch.median(output, dim=(- 1))
```

```
RuntimeError: dims < 25 INTERNAL ASSERT FAILED at "/opt/conda/conda-bld/pytorch_1666249970276/work/aten/src/ATen/cuda/detail/TensorInfo.cuh":53, please report a bug to PyTorch
```

### Versions

torch: 1.13.0
cuda: 11.6

cc @ngimel

### PR DESCRIPTION
**Pull request by cnemri**
Fixes #88808
Replaced

`AT_ASSERT(dims < MAX_TENSORINFO_DIMS) `
in aten/src/ATen/cuda/detail/TensorInfo.cuh by

```
  data = p;
  dims = dim;
  TORCH_CHECK(dims < MAX_TENSORINFO_DIMS, "CUDA Tensors cannot have more than 25 dimensions");
    }

```

In : torch/testing/_internal/common_methods_invocations.py

```
def error_inputs_median(op_info, device, **kwargs):
    x = torch.tensor([[[[[[[[[[[[[[[[[[[[[[[[[nan],
                               [nan]]]]]]]]]]]]]]]]]]]]]]]]], device=device)
    if device=='cuda':
        yield ErrorInput(SampleInput(x, kwargs=dict(dim=(-1))),
                        error_type=RuntimeError,
                        error_regex='CUDA Tensors cannot have more than 25 dimensions')
    else:
        return

```
And

```
    OpInfo('median',
           ...
           error_inputs_func=error_inputs_median,
           ...

```

### CODE REVIEW

**Comment by cnemri** [comment]
@ngimel @mruberry I've updated the code as suggested and recreated the PR to avoid the EasyCLA issue.

**Comment by ngimel** [comment]
You've accidentally updated submodules, can you please remove those updates?

**Comment by ngimel** [comment]
Otherwise, looks good, let's see what tests say

**Comment by cnemri** [comment]
@ngimel thanks for your feedback. I was not sure how to revert submodule changes, I had to do a hard reset and recommit changes over again. I am having a labeler worfklow run being cancelled due to `Canceling since a higher priority waiting request for 'Labeler-91273-false' exists `

**Comment by cnemri** [comment]
Hi team the build workflow is currently pending approval. Not very familiar with the process, is it because of this failing Labeler step (screenshot below) ? Or just because someone from the community should approve the PR so it goes through the full workflow?

image

**Comment by ngimel** [comment]
Approved the workflows

**Comment by cnemri** [comment]
Hi team, the build is over. 10 checks are failing because of the same error
`++ python3 .github/scripts/get_workflow_job_id.py 3756666170 i-0c4ac3fd428721882
Traceback (most recent call last):
  File ".github/scripts/get_workflow_job_id.py", line 48, in <module>
    jobs = response.json()["jobs"]
KeyError: 'jobs'
`
The latter seems to be unrelated to my changes.

**Comment by ngimel** [comment]
Yeah, test failures are due to SEV, see the message in the bot comment: 
```
There is 1 active merge blocking SEVs. Please view them below:

(merge blocking) [Many CI workflows are failing with KeyError: 'jobs'](https://hud.pytorch.org/pytorch/pytorch/issues/91332)
```
I'll rerun the tests when the SEV is over, thank you for the fix!

**Review by ngimel — APPROVED** [review]

**Comment by cnemri** [comment]
Sorry I clicked by mistake and removed the second reviewer … let me know if I can do anything about it

**Comment by ngimel** [comment]
np, don't worry

**Comment by cnemri** [comment]
Happy new year @ngimel and @mruberry how do I know if the SEV is over?

**Pull request closed without merging**
